core/src/main/scala/cats/Apply.scala

```diff
@@ -45,6 +37,9 @@ trait Apply[F[_]] extends Functor[F] with ApplyArityFunctions[F] { self =>
       def F: Apply[F] = self
       def G: Apply[G] = GG
     }
+
+  override def product[A, B](fa: F[A], fb: F[B]): F[(A, B)] = ap(fb)(map(fa)(a => b => (a, b)))
```

Comment:
Ok but then I have to update every place in the code where we create instances of `Apply` to implement `product` instead of `ap`. Are you ok with that?